lib/bolt/node/ssh.rb

```diff
@@ -34,6 +44,14 @@ def connect
                                   end
       options[:timeout] = @connect_timeout if @connect_timeout
 
+      if defined?(UNIXSocket) && UNIXSocket && ENV['SSH_AUTH_SOCK'].to_s.empty?
+        @logger.debug { "Disabling use_agent in net-ssh: ssh-agent is not available" }
+        options[:use_agent] = false
+      elsif Gem.win_platform? && Win.FindWindow('Pageant', 'Pageant').to_i == 0
```

Comment:
FWIW - checking for a specific putty process (by looking for its Window) is probably not a great heuristic to use.  I'm sure it won't work with git bash for instance, or other tools that may be part of Cygwin.

It seems like there should be something more reliable that can identify a running ssh-agent on Windows.

Reply:
Interestingly, posh-git does a similar thing (but looks for the process, not the window name, which might not work).

But it also does a number of additional things - https://github.com/dahlbyk/posh-git/blob/2ecf05a55b9cacb2e9bcc20321cccd983e25a428/src/GitUtils.ps1#L395-L415

Reply:
This is what net-ssh does I don't think there is much value in us doing better than net-ssh
https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/authentication/pageant.rb#L403

Reply:
True, but I specifically modeled the code in net-ssh in an attempt to symmetrically unwind its `use_agent` setting.

I still prefer https://github.com/puppetlabs/bolt/pull/240 which would allow bolt users to set `use_agent`, rather than hard code the default of `use_agent=true` and import these implementation details from net-ssh into bolt.

Reply:
Yeah, understand your point @adreyer ... haven't looked at net-ssh code too closely, but is there not a way to call to one of its helpers? I see `Net::SSH::Authentication:: Pageant::Socket.new` will raise if there's no Putty based ssh-agent. Of course, that won't help for others...

If we're relying on Net::SSH anyhow, seems like the best place to have a check is in that library that we can rely on?

Reply:
Oh ...

I've tried to rescue the Net::SSH:: Authentication::AgentNotAvailable
error within lib/bolt/node/ssh.rb, similar to what Nick Lewis
suggested in Jira. No success, and might not capture the error message anyway.

I've looked for something more 'public' in Net-SSH to use. Nothing
found, other than allowing the bolt user to set 'net-ssh::use_agent'
as per PR 240.